[core] Remove `createStyles` from `@material-ui/core` (#26018)
mui-org/material-ui: docs/src/pages/guides/migration-v4/migration-v4.md

}));
```

> -The `createStyles` function from `@material-ui/core/styles` is removed in favor of the one exported from `@material-ui/styles`. It is necessary for removing the dependency to `@material-ui/styles` in the core package.

Reviewer: Move would be more accurate?

```suggestion
-The `createStyles` function from `@material-ui/core/styles` was moved to the one exported from `@material-ui/styles`. It is necessary for removing the dependency to `@material-ui/styles` in the core package.
```
